It replaces the timespamp in the loglines
@400000004a9b768834d8118c status: local 0/500 remote 4/299
in human readable time
Sat Dec 19 03:02:51 2009  status: local 0/500 remote 0/299

More info here
https://web.archive.org/web/20120530042120/http://blog.endersys.com/2009/12/qmail-multilog-localtime-patch-for-daemontools-0-76/

diff -ruN daemontools-0.76-original/src/multilog.c daemontools-0.76/src/multilog.c
--- daemontools-0.76-original/src/multilog.c	2018-04-06 21:09:12.602802333 +0200
+++ daemontools-0.76/src/multilog.c	2020-04-16 16:26:51.713168982 +0200
@@ -516,7 +516,8 @@
       }
       if (!linelen)
         if (flagtimestamp) {
-          timestamp(line);
+          /* timestamp(line); */
+          timestamplocal(line);
           line[25] = ' ';
           linelen = 26;
         }
diff -ruN daemontools-0.76-original/src/timestamp.c daemontools-0.76/src/timestamp.c
--- daemontools-0.76-original/src/timestamp.c	2001-07-12 18:49:49.000000000 +0200
+++ daemontools-0.76/src/timestamp.c	2020-04-16 16:22:28.160628649 +0200
@@ -1,7 +1,13 @@
+#include <sys/types.h>
+#include <time.h>
+#include <sys/time.h>
+#include <unistd.h>
 #include "taia.h"
 #include "timestamp.h"
 
 static char hex[16] = "0123456789abcdef";
+time_t lt;
+struct tm *t;
 
 void timestamp(char s[TIMESTAMP])
 {
@@ -18,3 +24,11 @@
     s[i * 2 + 2] = hex[nowpack[i] & 15];
   }
 }
+
+void timestamplocal(char s[TIMESTAMP])
+{
+      lt = time(NULL);
+      t = localtime(&lt);
+      asctime_r(t, s);
+      s[24] = ' ';
+}
diff -ruN daemontools-0.76-original/src/timestamp.h daemontools-0.76/src/timestamp.h
--- daemontools-0.76-original/src/timestamp.h	2001-07-12 18:49:49.000000000 +0200
+++ daemontools-0.76/src/timestamp.h	2020-04-16 16:22:28.160628649 +0200
@@ -4,5 +4,6 @@
 #define TIMESTAMP 25
 
 extern void timestamp(char *);
+extern void timestamplocal(char *);
 
 #endif
